src/wasm-binary.h

```diff
@@ -1398,12 +1398,14 @@ class WasmBinaryBuilder {
                           Address defaultIfNoMax);
   void readImports();
 
-  // The signatures of each function, given in the function section
-  std::vector<Signature> functionSignatures;
+  // The signatures of each function, including imported functions, given in the
+  // import and function sections. Store HeapTypes instead of Signatures because
+  // reconstructing the HeapTypes from the Signatures is expensive.
+  std::vector<HeapType> functionSignatures;
```

Comment:
how about renaming this to `functionTypes`?

Reply:
Also note the expanded comment calling out that this now includes imported functions as well.